pull/github.go

```diff
@@ -26,6 +26,10 @@ import (
 )
 
 const (
+	// GitHubAppID is the GitHub Actions App ID. When a check suite is created
+	// by GitHub Actions, its `app.id` field is set to this value.
+	GitHubAppID = 15368
```

Comment:
If we have to use this (which might not be necessary if we switch to the REST API), it will need to be configurable. For example, on one of our internal GitHub Enterprise instances, the App ID for the `github-actions` app is 152.

Another alternative might be to filter based on the app name (which should be consistent) instead of the ID.

Reply:
Ah, cheers for the info. If we use the `slug` field - which is `github-actions` for github.com - would that work in your environment too?

Reply:
Yeah, when I checked before it looked like the slug was consistent across both github.com and GitHub Enterprise

Reply:
[The API wants an ID though](https://pkg.go.dev/github.com/google/go-github/v63/github#ListCheckSuiteOptions), so I'm going to have to resolve it. Since that's an API call and the value isn't going to change, I should cache it for the lifetime of the whole program. Do you have a preference for where/how we should do that?

Reply:
Not sure if this will do everything else you needed, but if we switch to the [List workflow runs](https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28#list-workflow-runs-for-a-repository) REST API, I think it bypasses this problem since we know it's only returning workflow details from actions and it includes the paths and the conclusions.

If you do end up needing to cache the ID, I think you could do it at server startup and pass it in to the base handler. Or maybe we make it a server configuration option at that point and operators will have to look it up before using this feature.